lib/metadataTypes/Automation.js

```diff
@@ -709,6 +710,172 @@ class Automation extends MetadataType {
+                json.r__folder_Path
+                    ? json.r__folder_Path
+                    : '_Hidden! Could not find folder with ID ' + json.categoryId + '_'
+            }/\n\n`;
+        const automationType = { scheduled: 'Schedule', triggered: 'File Drop' };
+        output += `**Started by:** ${
+            automationType[json.type] ? automationType[json.type] : 'Not defined'
+        }\n\n`;
+        const tz =
+            this.definition.timeZoneDifference[
+                this.definition.timeZoneMapping[json.schedule.timezoneName]
+            ];
+
+        if (json.schedule?.icalRecur) {
+            output += `**Schedule:**\n\n`;
+            output += `* Start: ${json.schedule.startDate.split('T').join(' ')} ${tz}\n`;
+            output += `* End: ${json.schedule.endDate.split('T').join(' ')} ${tz}\n`;
+            output += `* Timezone:  ${json.schedule.timezoneName}\n`;
+
+            const ical = {};
+            json.schedule.icalRecur.split(';').forEach((item) => {
+                const temp = item.split('=');
+                ical[temp[0]] = temp[1];
+            });
+            const frequency = ical.FREQ.slice(0, -2).toLowerCase();
+
+            output += `* Recurrance: every ${ical.INTERVAL > 1 ? ical.INTERVAL : ''} ${
+                frequency === 'dai' ? 'day' : frequency
+            }${ical.INTERVAL > 1 ? 's' : ''} ${ical.COUNT ? `for ${ical.COUNT} times` : ''}\n`;
+            output += '\n';
+        } else if (json.schedule) {
+            output += `**Schedule:** Not defined\n\n`;
+        }
+
+        let tableSeparator = '';
+        const row1 = [];
+        tabled[0].forEach((column) => {
+            row1.push(
+                `| ${column.title}${column.description ? `<br>_${column.description}_` : ''} `
+            );
+            tableSeparator += '| --- ';
+        });
+        output += row1.join('') + `|\n${tableSeparator}|\n`;
+        for (let i = 1; i < tabled.length; i++) {
+            tabled[i].forEach((field) => {
+                if (field) {
+                    output += `| _${field.i}: ${field.type}_<br>${field.name} `;
+                } else {
+                    output += '| - ';
+                }
+            });
+            output += '|\n';
+        }
+        return output;
+    }
+    /**
+     * Saves json content to a html table in the local file system. Will create the parent directory if it does not exist.
+     * The json's first level of keys must represent the rows and the secend level the columns
+     *
+     * @private
+     * @param {string} directory directory the file will be written to
+     * @param {string} filename name of the file without '.json' ending
+     * @param {TYPE.AutomationItem} json dataextension.columns
+     * @param {'html'|'md'} mode html or md
+     * @returns {Promise.<boolean>} Promise of success of saving the file
+     */
+    static async _writeDoc(directory, filename, json, mode) {
+        if (!File.existsSync(directory)) {
```

Comment:
prefer async

Reply:
prefer [ensureDir](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir.md). Also would be good to put this in the writeToFile method, to avoid having this in a bunch of different places